builder/build.go

```diff
@@ -72,22 +69,23 @@ func Build(pkgName, outpath string, config *compileopts.Config, action func(stri
 
 	// Optimization levels here are roughly the same as Clang, but probably not
 	// exactly.
+	errs = nil
 	switch config.Options.Opt {
 	case "none:", "0":
-		err = c.Optimize(0, 0, 0) // -O0
+		errs = c.Optimize(0, 0, 0) // -O0
 	case "1":
-		err = c.Optimize(1, 0, 0) // -O1
+		errs = c.Optimize(1, 0, 0) // -O1
 	case "2":
-		err = c.Optimize(2, 0, 225) // -O2
+		errs = c.Optimize(2, 0, 225) // -O2
 	case "s":
-		err = c.Optimize(2, 1, 225) // -Os
+		errs = c.Optimize(2, 1, 225) // -Os
 	case "z":
-		err = c.Optimize(2, 2, 5) // -Oz, default
+		errs = c.Optimize(2, 2, 5) // -Oz, default
 	default:
-		err = errors.New("unknown optimization level: -opt=" + config.Options.Opt)
+		errs = []error{errors.New("unknown optimization level: -opt=" + config.Options.Opt)}
 	}
-	if err != nil {
-		return err
+	if len(errs) != 0 {
```

Comment:
```suggestion
	if len(errs) > 0 {
```
This would be slightly more readable as `> 0`, but this is kinda minor.

Reply:
To me it's exactly equally readable, perhaps `!= 0` is even more readable as it test for a non-zero error (`errs != nil` would also have worked but theoretically `errs != nil && len(errs) == 0` could be true). But on the other hand, I can see where you're coming from.